src/Mvp.Selections.Data/Repositories/UserRepository.cs

```diff
@@ -156,6 +156,16 @@ public async Task<IList<User>> GetAllForRolesReadOnlyAsync(IEnumerable<Guid> rol
             .ToListAsync();
     }
 
+    public async Task<bool> UserHasTitleForYearAsync(Guid userId, int year)
+    {
+        return await Context.Applications
+            .Where(a => a.Applicant.Id == userId &&
+                        a.Selection.Finalized &&
+                        a.Selection.Year == year &&
+                        a.Titles.Any())
+            .AnyAsync();
+    }
```

Comment:
Does this call need to be Async?

If you look at the `DoesUserExist(string identifier)` method above, changing it to not being async would allow us to match the simpler implementation that uses. You can then also directly use an `Any()` call instead of needing to chain `Where().AnyAsync()`